src/LazyLoadImage/LazyLoadImage.tsx

```diff
@@ -34,20 +42,43 @@ export default function LazyLoadImage({
       { threshold: 0.1 }
     );
     if (imgRef.current) {
+      const { offsetWidth, offsetHeight } = imgRef.current;
+      setDimensions({ height: offsetHeight, width: offsetWidth });
       observer.observe(imgRef.current);
     }
     return () => observer.disconnect();
   }, []);
 
+  // a variable to decide whether to show the Skeleton, or the image (or the alt text in case an error happened during loading)
+  const showSkeleton = isVisible && !isLoaded && !hasError;
+
+  // deconstruct ImgProps and the style, to apply the needed style.display prop for the image to lazy load without issues
+  const { style, ...restImgProps } = ImgProps || {};
+  const { display, ...restStyle } = style || {};
+
   // render the image
   return (
-    <img
-      alt={alt || ""}
-      ref={imgRef}
-      loading="lazy"
-      src={isVisible ? src : undefined}
-      style={{ display: isVisible ? "block" : "none" }}
-      {...ImgProps}
-    />
+    <>
+      {showSkeleton && (
+        <Skeleton
+          sx={{
+            height: dimensions.height,
+            width: dimensions.width
+          }}
+        />
+      )}
+      <img
+        alt={hasError ? alt : ""}
+        ref={imgRef}
+        src={isVisible ? src : undefined}
+        style={{
+          ...restStyle,
+          display: showSkeleton ? "none" : "block"
```

Comment:
another difference compared to how we are styling in `Infographic`. In Infographic don't need to worry about overwriting styling, as its styling is hard-coded.

Since this is a wrapper, styling comes from the props. I decontructed styling and use my own logic to make sure the display prop is doing what we need.